Cilsil/Cil/Parsers/LeaveParser.cs

```diff
@@ -19,7 +19,8 @@ internal class LeaveParser : InstructionParser
 
                     state.AppendToPreviousNode = false;
 
-                    if (targetTrue.Offset != nextInstruction.Offset)
+                    if (targetTrue.Offset != nextInstruction.Offset &&
```

Comment:
As I mentioned in another comment -- the problem with this logic is that if the leave routes to a CIL instruction which consumes from the stack (and I have not seen validation that this cannot occur), then this could cause issues. The stack would then consume the return expression you pushed onto it -- this would then be an inaccurate representation of the intended CIL behavior.

Reply:
As the condition at line 22 shows, this logic only applies when its next instruction is a catch/finally block. As I mentioned in another comment that from Java sample code the only purpose I saw is to store the returned variable, which is to be loaded to construct the very first catch/finally block node like the example shown below:

```
13: Preds: [] Succs: [11, 12] EXN: [3] 
n$5=*&return:int [line 23];
*&return:int=null [line 23];
n$6=_fun___unwrap_exception(n$5:int) [line 23];
```